### PR TITLE
fix: tsfresh abs features and change quantiles

### PR DESCRIPTION
**Pull request by MathieuCayssol**

### CODE REVIEW

File: functime/feature_extraction/tsfresh.py [hunk 1]
```diff
@@ -49,12 +49,12 @@ def absolute_maximum(x: TIME_SERIES_T) -> FLOAT_EXPR:
     float | Expr
     """
     if isinstance(x, pl.Series):
-        return np.max(np.abs([x.min(), x.max()]))
```
Comment (abstractqqq) [inline functime/feature_extraction/tsfresh.py:52]: I wrote it this way using NumPy for the following reason:

Calling abs() on the entire series will trigger a temporary allocation of a new series as long as the original x. By only computing x.min() and x.max() and taking the absolute value, you avoid this allocation. On larger data, this improves performance by a lot.

You can verify mathematically the max of the absolute value must be either |x.min()| or |x.max()|, so creating an entire series by x.abs() is unnecessary.

The old implementation is better.

File: functime/feature_extraction/tsfresh.py [hunk 2]
```diff
@@ -356,8 +354,14 @@ def c3(x: TIME_SERIES_T, n_lags: int) -> FLOAT_EXPR:
...
     """
     twice_lag = 2 * n_lags
+    if isinstance(x, pl.Series):
+        if twice_lag >= x.len():
+            return np.nan
+        else:
+            return (x * x.shift(n_lags) * x.shift(twice_lag)).sum() / (x.len() - twice_lag)
+    else:
     # Would potentially be faster if there is a pl.product_horizontal()
-    return (x * x.shift(n_lags) * x.shift(twice_lag)).sum() / (x.len() - twice_lag)
```
Comment (abstractqqq) [inline functime/feature_extraction/tsfresh.py:360]: For series input, there is no .mul(), only _ _ mul _ _. 

Using * works for both series and expression because both has _ _ mul_ _.

File: functime/feature_extraction/tsfresh.py [hunk 3]
```diff
@@ -424,13 +428,13 @@ def cid_ce(x: TIME_SERIES_T, normalize: bool = False) -> FLOAT_EXPR:
...
     if normalize:
-        y = (x - x.mean()) / x.std()
+        y = (x - x.mean()) / x.std(ddof=0)
     else:
         y = x
 
     if isinstance(x, pl.Series):
-        diff = np.diff(x)
-        return np.sqrt(np.dot(diff, diff))
+        diff = y - y.shift(periods=-1)
```
Comment (abstractqqq) [inline functime/feature_extraction/tsfresh.py:436]: Using diff is faster because y.shift creates another Series, while diff just takes two values at a time... 

Using .dot is faster because there is no intermediate Series being created. It is just accumulation on a single value.

File: functime/feature_extraction/tsfresh.py [hunk 4]
```diff
@@ -467,7 +471,11 @@ def count_above_mean(x: TIME_SERIES_T) -> INT_EXPR:
     -------
     int | Expr
     """
-    return (x > x.mean()).sum()
+    if isinstance(x, pl.Series):
+        y = x.cast(pl.Float64)
```
Comment (abstractqqq) [inline functime/feature_extraction/tsfresh.py:475]: Why do we need this cast?

File: functime/feature_extraction/tsfresh.py [hunk 5]
```diff
@@ -197,8 +197,6 @@ def autocorrelation(x: TIME_SERIES_T, n_lags: int) -> FLOAT_EXPR:
     """
     if n_lags < 0:
         raise ValueError("Input `n_lags` must be >= 0")
-    elif n_lags == 0:
-        return 1.0
```
Comment (abstractqqq) [inline functime/feature_extraction/tsfresh.py:201]: Why did you remove this? 0 lag autocorrelation should have a shortcut...

**Review by abstractqqq — CHANGES_REQUESTED** [review]

**Review by topher-lo — COMMENTED** [review]
Waiting for @MathieuCayssol to respond to @abstractqqq

**Comment by abstractqqq** [comment]
> Waiting for @MathieuCayssol to respond to @abstractqqq

I think the problem is that he didn't pull the changes I made in tsfresh.py...

Everything in unit test looks good. Only the things in tsfresh.py..

**Comment by MathieuCayssol** [comment]
I did pull your changes from the `main` branch after your merge in `main` but kept the pure polars changes I've made. I will revert back to your numpy implementation after our alignement.

File: functime/feature_extraction/tsfresh.py (hunk 2 above)
Comment (MathieuCayssol) [reply functime/feature_extraction/tsfresh.py:360]: As I want to avoid the user to face this message while using this function in eager mode:

`ZeroDivisionError: division by zero`

I need to perform `twice_lag >= x.len()` and therefore I need to distinguish eager and lazy.

Ofc, I can do this as well:

```
twice_lag = 2*n_lags
if isinstance(x, pl.Series):
    if twice_lag >= x.len():
        return np.nan
return ((x *(x.shift(n_lags))* (x.shift(twice_lag))).sum()) / ((x.len() - twice_lag))
```

File: functime/feature_extraction/tsfresh.py (hunk 3 above)
Comment (abstractqqq) [reply functime/feature_extraction/tsfresh.py:436]: I believe we agreed that it is ok to use NumPy for eager case. Let's revert this too.

**Review by abstractqqq — CHANGES_REQUESTED** [review]

File: functime/feature_extraction/tsfresh.py (hunk 3 above)
Comment (MathieuCayssol) [reply functime/feature_extraction/tsfresh.py:436]: Yes, I know, I was busy on something else.

**Review by abstractqqq — APPROVED** [review]
